### ISSUE TITLE
Cryptomator Folders Not Showing on iOS

### ISSUE THREAD

**Issue by skarvall** [cryptomator/ios#338]
### Please agree to the following

- [X] I have searched [existing issues](https://github.com/cryptomator/ios/issues?q=) for duplicates
- [X] I agree to follow this project's [Code of Conduct](https://github.com/cryptomator/ios/blob/develop/.github/CODE_OF_CONDUCT.md)

### Summary

(Some) Cryptomator Folders/Files Not Showing on iOS

### - Tested Environment:

**- Devices:**

iPhone 15 PRO | iOS Version 17.3.1
iPhone XS | iOS Version 17.4 (beta 4)
Cryptomator Version: 2.5.0
Cryptomator Version: 2.5.0 (Testflight)

### Cloud Type

iCloud Drive

### Steps to Reproduce

**- Description:**

Certain files in Cryptomator folders created on MacOS or iOS for iCloud are no longer showing on iOS devices. When attempting to access these folders, the following error message is displayed: "Content Unavailable: The folder contents could not be displayed because of an unknown error."

### Expected Behavior

All folders and files should show.

### Actual Behavior

Some files and folders are not accessable.

### Reproducibility

Always

### Anything else?

**- Steps Taken:**

Unmounting and mounting the vaults had no effect.
Reinstalling the app had no effect.
Rebooting the devices had no effect.
Re-adding a new folder from MacOS.
Re-adding a new folder from iOS.
Switching between WIFI and cellular had no effect.

**- Possible Bypass:**

It is possible to press "Try again" under the aforementioned error message. While doing so, it is possible to simultaneously press the 'three dots menu' in the top right, and then create a 'New Folder'. After returning to the parent folder and then going back to the folder in question, the files are visible again.

**- Here is an example:**

[https://drive.proton.me/urls/KTYC8SZHG8#8mjIpavH22rH](https://drive.proton.me/urls/KTYC8SZHG8#8mjIpavH22rH)

The password is the name of the software - all lowercase.

Please let me know if I missed an obvious solution to this issue.

**Comment by tobihagemann** [comment]
Thank you for your bug report. It's possibly a duplicate of #316. Something with iOS 17 has changed and there is this "Resource deadlock avoided" error that's causing problems in some cases. Unfortunately, we weren't able to reproduce this issue yet and there is not any information on the internet about this. Can you check your log files if you're having this same error? I'm not closing this as duplicate yet because I find your workaround very interesting. But check out the other issue, which mentions a different workaround as well.

**Comment by skarvall** [comment]
Hi,

Thank you for the prompt feedback. I did find the same "Resource deadlock avoided" error present in the log files. It appears to be a duplicate of #316 .

Unfortunately, my vault is simply too large to use the download bypass, but my "folder solution" seems to be doing the trick so far.

I have attached all my log files (.zip) from one of my iPhones. It's not much, but maybe they are of interest.

[Logs.zip](https://github.com/cryptomator/ios/files/14594830/Logs.zip)
